pkg/ingester/ingester_v2.go

```diff
@@ -847,6 +851,37 @@ func (i *Ingester) v2Push(ctx context.Context, req *cortexpb.WriteRequest) (*cor
 				return copiedLabels
 			})
 		}
+
+		// app.AppendExemplar currently doesn't create the series, it must
+		// already exist.  If it does not then drop.
+		if ref == 0 && len(ts.Exemplars) > 0 {
+			updateFirstPartial(func() error {
+				return wrappedTSDBIngestExemplarErr(errors.New("exemplars not ingested because series not already present"),
```

Comment:
I think not ingesting the exemplar if we don't already know about the series is the correct thing to do, but I'm not sure what kind of error message we want to return to users here.

Reply:
`Do we want to return an error at all? Metadata ingestion is best-effort and we never return any error about it. Should exemplars ingestion be a best-effort as well?` 

We should IMO, exemplars are more useful/important than metadata. Exemplar ingestion could be best-effort right now for the in-memory storage, though it shouldn't be forever when we eventually have a better storage implementation. FWIW what Marty has done here is essentially exactly what would happen in Prometheus' exemplar storage if you tried to store an exemplar for a series: https://github.com/prometheus/prometheus/blob/main/tsdb/head.go#L1326-L1345